syntax = "proto3";

package bookstore;

// Definição do serviço de Catálogo de Livros
service CatalogService {
    rpc GetBookInfo (BookRequest) returns (BookInfoResponse);
    rpc ListBooks (Empty) returns (BookListResponse);
}

// Mensagens utilizadas pelo serviço
message BookRequest {
    string b_id = 1;
}

message BookInfoResponse {
    string b_id = 1;
    string title = 2;
    string author = 3;
    int32 year = 4;
    int32 stock = 5;
    double price = 6;
    string erro = 7;
}

message Empty {}

message Book {
    string b_id = 1;
    string title = 2;
    string author = 3;
    int32 year = 4;
    int32 stock = 5;
    double price = 6;
}

message BookListResponse {
    repeated Book books = 1;
}
